src/rust/kafka/src/lib.rs

```diff
@@ -0,0 +1,279 @@
+use std::{
+    fmt::Display,
+    marker::PhantomData,
+};
+
+use futures::{
+    stream::{
+        Stream,
+        StreamExt,
+    },
+    FutureExt,
+    TryFutureExt,
+    TryStreamExt,
+};
+use rdkafka::{
+    config::ClientConfig,
+    consumer::stream_consumer::StreamConsumer,
+    error::KafkaError,
+    producer::{
+        FutureProducer,
+        FutureRecord,
+    },
+    util::Timeout,
+    Message,
+};
+use rust_proto_new::{
+    SerDe,
+    SerDeError,
+};
+use thiserror::Error;
+
+//
+// Kafka configurations
+//
+
+#[derive(Error, Debug)]
+pub enum ConfigurationError {
```

Comment:
Almost forgot to mention, I recommend `#[non_exhaustive]` for things like pub error types:  https://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute.

Reply:
Oh cool TIL. Will add.